icu4c/source/i18n/gregoimp.cpp

```diff
@@ -130,17 +130,15 @@ void Grego::dayToFields(double day, int32_t& year, int32_t& month,
     UBool isLeap = isLeapYear(year);
     
     // Gregorian day zero is a Monday.
-    dow = (int32_t) uprv_fmod(day + 1, 7);
-    dow += (dow < 0) ? (UCAL_SUNDAY + 7) : UCAL_SUNDAY;
+    dow = ((int32_t) day + 1) % 7;
+    dow += UCAL_SUNDAY + 7*(dow < 0);
 
     // Common Julian/Gregorian calculation
-    int32_t correction = 0;
-    int32_t march1 = isLeap ? 60 : 59; // zero-based DOY for March 1
-    if (doy >= march1) {
-        correction = isLeap ? 1 : 2;
-    }
+    int32_t march1 = 59 + isLeap; // zero-based DOY for March 1
```

Comment:
Not crazy about using `isLeap` as an integer this way; I've seen too many cases where somebody used a value other than 1 for `true`, and I also don't think it's as clear.  Is there really a performance advantage to what you're doing here?

Reply:
Okay, apparently `UBool` is actually `int8_t`. That typedef should probably be changed to `bool` to get stronger guarantees based on the C++ spec about the allowed values, as this is a possibly hidden source of bugs. Given the current situation, I'll revert the changes assuming true is 1.

Reply:
As I remember it, there were compatibility issues with changing the definition of `UBool` to just be `bool`, but we've started using `bool` internally in new code (I think it's just the API where we're stuck with it).